### PR TITLE
Make JsonbType generic

### LINKED ISSUE #9165: Make JsonbType generic
### Description

Newly added `JsonbType` [class](https://github.com/keycloak/keycloak/blob/main/model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/hibernate/jsonb/JsonbType.java) uses hardcoded classes for JpaClient store. 

To be able to use it with other stores it needs to be refactored. 

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by vramik**
Closes #9165

### CODE REVIEW

File: model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/hibernate/jsonb/JsonbType.java [hunk 1]
```diff
@@ -148,9 +154,14 @@ private Object extractJson(CallableStatement statement, String name) throws SQLE
...
 
-    private static class JsonbJavaTypeDescriptor extends AbstractTypeDescriptor<Object> {
+    private static class JsonbJavaTypeDescriptor extends AbstractTypeDescriptor<Object> implements DynamicParameterizedType {
 
-        private static final JsonbJavaTypeDescriptor INSTANCE = new JsonbJavaTypeDescriptor();
+        private Class valueType;
+
+        @Override
+        public void setParameterValues(Properties parameters) {
+            valueType = ((ParameterType) parameters.get(PARAMETER_TYPE)).getReturnedClass();
```
Comment (sguilhen) [inline model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/hibernate/jsonb/JsonbType.java:163]: Where is PARAMETER_TYPE defined?

Comment (vramik) [reply model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/hibernate/jsonb/JsonbType.java:163]: There is the declaration: https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/usertype/DynamicParameterizedType.java#L22

**Review by sguilhen — DISMISSED** [review]
Thanks @vramik , this looks good to me!

**Review by hmlnarik — APPROVED** [review]
Thanks for the fix!

**Pull request merged**